cmd/skupper/skupper_link.go

```diff
@@ -0,0 +1,185 @@
+				if err != nil {
+					return fmt.Errorf("Failed to create connection: %w", err)
+				} else {
+					if siteConfig.Spec.IsEdge {
+						fmt.Printf("Skupper configured to connect to %s:%s (name=%s)\n",
+							secret.ObjectMeta.Annotations["edge-host"],
+							secret.ObjectMeta.Annotations["edge-port"],
+							secret.ObjectMeta.Name)
+					} else {
+						fmt.Printf("Skupper configured to connect to %s:%s (name=%s)\n",
+							secret.ObjectMeta.Annotations["inter-router-host"],
+							secret.ObjectMeta.Annotations["inter-router-port"],
+							secret.ObjectMeta.Name)
+					}
+				}
+			} else {
+				// create the secret, site-controller will do the rest
+				secret, err := cli.ConnectorCreateSecretFromFile(context.Background(), args[0], connectorCreateOpts)
+				if err != nil {
+					return fmt.Errorf("Failed to create connection: %w", err)
+				} else {
+					if siteConfig.Spec.IsEdge {
+						fmt.Printf("Skupper site-controller configured to connect to %s:%s (name=%s)\n",
+							secret.ObjectMeta.Annotations["edge-host"],
+							secret.ObjectMeta.Annotations["edge-port"],
+							secret.ObjectMeta.Name)
+					} else {
+						fmt.Printf("Skupper site-controller configured to connect to %s:%s (name=%s)\n",
+							secret.ObjectMeta.Annotations["inter-router-host"],
+							secret.ObjectMeta.Annotations["inter-router-port"],
+							secret.ObjectMeta.Name)
+					}
+				}
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&connectorCreateOpts.Name, flag, "", "", "Provide a specific name for the connection (used when removing it with disconnect)")
+	cmd.Flags().Int32VarP(&connectorCreateOpts.Cost, "cost", "", 1, "Specify a cost for this connection.")
+
+	return cmd
+}
+
+var connectorRemoveOpts types.ConnectorRemoveOptions
+
+func NewCmdLinkDelete(newClient cobraFunc) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "delete <name>",
+		Short:  "Remove specified link",
+		Args:   cobra.ExactArgs(1),
+		PreRun: newClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			silenceCobra(cmd)
+			connectorRemoveOpts.Name = args[0]
+			connectorRemoveOpts.SkupperNamespace = cli.GetNamespace()
+			connectorRemoveOpts.ForceCurrent = false
+			err := cli.ConnectorRemove(context.Background(), connectorRemoveOpts)
+			if err == nil {
+				fmt.Println("Link '" + args[0] + "' has been removed")
+			} else {
+				return fmt.Errorf("Failed to remove link: %w", err)
+			}
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+var waitFor int
+
+func NewCmdLinkStatus(newClient cobraFunc) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "status [<connection-name>]",
+		Short:  "Check whether a link to another Skupper site is active",
+		Args:   cobra.MaximumNArgs(1),
+		PreRun: newClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			silenceCobra(cmd)
+
+			var connectors []*types.ConnectorInspectResponse
+			connected := 0
+
+			linkName := "all"
+			if len(args) == 1 {
+				linkName = args[0]
+			}
+
+			if linkName == "all" {
```

Comment:
Maybe you can simply remove all this code and instead of `if linkName == "all"`,
just do a `if len(args) == 0` in order to list all links, then at the else you get linkName from args[0].

Reply:
the reason is that, this code is also executed for the deprecated command  "check-connection" and for that command, args[0] may be "all", so here I am supporting args[0] == all (for the deprecated command) and args[0]= "empty" for the new command

Reply:
Sounds good then.